### PR TITLE
Feat: Improving Memory and Looping

### PR DESCRIPTION
**Pull request by kamikazebr**
- feat: SKIP ACTION in case loop based in tool args
- feat: Try final answer after firtst task done
- feat: rollback full episodes instead summary in reasoning
- feat: improv path for files
- fix: clean ups and fix VSCode warnings
- feat: added depplake v2
- feat: backup subquestions and task list locally to speed up
- wip: insights from the past episodes when ran out limit tokens
- fix: typo
- prompt: final answer can return "I dont know" to continue after task done

### CODE REVIEW

File: src/llm/reason/prompt.py [hunk 1]
```diff
@@ -46,6 +46,33 @@
 {related_past_episodes}
 
 
+[YOUR TASK]
+You are given the following task:
+{task}
+
+[TOOLS USAGE]
+You can ONLY ONE TOOL at a time and only tool that are listed below. Remember to use task_complete tool to mark the task as done.
```
Comment (TjitsevdM) [inline src/llm/reason/prompt.py:54]: Adjust the prompt to:
You can ONLY USE ONE TOOL at a time and only use tools that are listed below

File: src/llm/summarize/prompt.py [hunk 2]
```diff
@@ -51,7 +51,8 @@
 {related_past_episodes}
 
 [INSTRUCTION]
-Using above [RESULTS OF COMPLETED TASKS], [RELATED KNOWLEDGE], and [RELATED PAST EPISODES], answer the [GOAL].
+ - Using above [RESULTS OF COMPLETED TASKS], [RELATED KNOWLEDGE], and [RELATED PAST EPISODES], answer the [GOAL].
+ - If not possible find the answer, return just nothing: 'I don't know'.
```
Comment (TjitsevdM) [inline src/llm/summarize/prompt.py]: Replace line 55 with:
- If you think performing [NEXT SUBTASK] will lead to a better answer, answer "continue"
- If it's not possible to find the answer, answer "I don't know"

**Review by TjitsevdM — CHANGES_REQUESTED** [review]
See my in code comments for prompt improvements. For the rest, everything looks good

File: src/llm/summarize/prompt.py (hunk 2 above)
Comment (kamikazebr) [reply src/llm/summarize/prompt.py]: What is [NEXT SUBTASK] ? We dont have it currently. Oh [NEXT POSSIBLE TASKS]? I see.

**Pull request merged**